api/types/types_paths.go

```diff
@@ -121,43 +191,88 @@ func (k fileKey) key() string {
 		return "log"
 	case Run:
 		return "run"
+	case TLS:
+		return "tls"
 	case LSX:
 		return "lsx"
 	}
 	return ""
 }
 
+func isLibstorageHomeSet() bool {
+	return libstorageHome != "/"
+}
+
 func (k fileKey) defaultVal() string {
+	if v, ok := k.isEnvVarSet(); ok {
+		return v
+	}
 	switch k {
 	case Home:
 		return libstorageHome
 	case Etc:
+		if isLibstorageHomeSet() {
+			return "/etc"
+		}
 		return "/etc/libstorage"
 	case Lib:
+		if isLibstorageHomeSet() {
+			return "/var/lib"
+		}
 		return "/var/lib/libstorage"
 	case Log:
+		if isLibstorageHomeSet() {
+			return "/var/log"
+		}
 		return "/var/log/libstorage"
 	case Run:
+		if isLibstorageHomeSet() {
+			return "/var/run"
+		}
 		return "/var/run/libstorage"
+	case TLS:
+		return "tls"
```

Comment:
Hi @codenrhoden,

Because of the function [`parent`](https://github.com/akutz/libstorage/blob/ac4fa0e8d091c51bedbd2bd166de191326cb4dd3/api/types/types_paths.go#L173):

```go
func (k fileKey) parent() fileKey {
	switch k {
	case TLS:
		return Etc
	case LSX:
		return Lib
	default:
		return Home
	}
}
```

Unless specified by an env var, the TLS path is relative to the Etc path.